charts/agent/templates/_deployment-cluster-events-config.tpl

```diff
@@ -186,6 +186,67 @@ processors:
+          - set(attributes["observe_transform"]["facets"]["succeededPods"], body["status"]["succeeded"])
+          - set(attributes["observe_transform"]["facets"]["readyPods"], body["status"]["ready"])
+          # spec
+          - set(attributes["observe_transform"]["facets"]["completions"], body["spec"]["completions"])
+          - set(attributes["observe_transform"]["facets"]["parallelism"], body["spec"]["parallelism"])
+          - set(attributes["observe_transform"]["facets"]["activeDeadlineSeconds"], body["spec"]["activeDeadlineSeconds"])
+          - set(attributes["observe_transform"]["facets"]["backoffLimit"], body["spec"]["backoffLimit"])
+      # Cron Job
+      - context: log
+        conditions:
+          - body["kind"] == "CronJob"
+        statements:
+          # spec
+          - set(attributes["observe_transform"]["facets"]["schedule"], body["spec"]["schedule"])
+          - set(attributes["observe_transform"]["facets"]["suspend"], "Active") where body["spec"]["suspend"] == false
+          - set(attributes["observe_transform"]["facets"]["suspend"], "Suspend") where body["spec"]["suspend"] == true
+          # status
+          - set(attributes["observe_transform"]["facets"]["activeJobs"], Len(body["status"]["active"]))
+      # Daemon Set
+      - context: log
+        conditions:
+          - body["kind"] == "DaemonSet"
+        statements:
+          # status
+          - set(attributes["observe_transform"]["facets"]["ready"], body["status"]["numberReady"])
+          - set(attributes["observe_transform"]["facets"]["desiredNumberScheduled"], body["status"]["desiredNumberScheduled"])
+          - set(attributes["observe_transform"]["facets"]["currentNumberScheduled"], body["status"]["currentNumberScheduled"])
+          - set(attributes["observe_transform"]["facets"]["updatedNumberScheduled"], body["status"]["updatedNumberScheduled"])
+          - set(attributes["observe_transform"]["facets"]["numberAvailable"], body["status"]["numberAvailable"])
+          - set(attributes["observe_transform"]["facets"]["numberUnavailable"], body["status"]["numberUnavailable"])
+          - set(attributes["observe_transform"]["facets"]["numberMisscheduled"], body["status"]["numberMisscheduled"])
+          # spec
+          - set(attributes["observe_transform"]["facets"]["updateStrategy"], body["spec"]["updateStrategy"]["type"])
+          - set(attributes["observe_transform"]["facets"]["maxUnavailable"], body["spec"]["updateStrategy"]["rollingUpdate"]["maxUnavailable"])
+          - set(attributes["observe_transform"]["facets"]["maxUnavailable"], 0) where attributes["observe_transform"]["facets"]["maxUnavailable"] == nil
+          - set(attributes["observe_transform"]["facets"]["maxSurge"], body["spec"]["updateStrategy"]["rollingUpdate"]["maxSurge"])
+          - set(attributes["observe_transform"]["facets"]["maxSurge"], 0) where attributes["observe_transform"]["facets"]["maxSurge"] == nil
```

Comment:
curious, why do you set it to 0 when the field doesn't exist?
(i.e. is 0 better than nil? if so, under what situation, and why?

Reply:
I had it nil, but then looking at our code we are already doing this thing for readyReplicas (Deployment and ReplicaSet). I think `nil` is better tho. I will revert this and remove the set to 0.